File: webapp/src/main/js/services/webrtcService.js
import WebrtcConfiguration from '../models/WebrtcConfiguration';

/**
 * WebRTC support.
 *
 * @author Alibaba Cloud
 */
export default {
    /**
     * Get the configuration of the WebRTC gateway and TURN/STUN server.
     *
     * @return {Promise.<WebrtcConfiguration>}
     */
    getConfiguration() {
        return new Promise((resolve, reject) => {
            fetch('/webrtc-configuration')
                .then(result => result.json())
                .then(
                    result => resolve(new WebrtcConfiguration(result)),
                    error => reject('Unable to get the WebRTC configuration: ' + JSON.stringify(error))
                );
        });
    }
};

File: webapp/src/main/java/com/alibaba/intl/livevideo/services/WebrtcService.java
package com.alibaba.intl.livevideo.services;

import com.alibaba.intl.livevideo.models.WebrtcConfiguration;

/**
 * WebRTC support.
 *
 * @author Alibaba Cloud
 */
public interface WebrtcService {

    /**
     * @return Configuration for the client to establish a WebRTC connection.
     */
    WebrtcConfiguration getConfiguration();

}
